app/scripts/HeatmapTiledPixiTrack.js

```diff
@@ -1026,6 +1026,19 @@ class HeatmapTiledPixiTrack extends TiledPixiTrack {
 
     this.renderingTiles.add(tile.tileId);
 
+    if (this.tilesetInfo.tile_size) {
+      if (tile.tileData.dense.length < this.tilesetInfo.tile_size) {
```

Comment:
Does that mean the server can send corrupt data? I like this fix but shouldn't we make sure the server sends proper data too? (Maybe there's another fix?)

Reply:
It's not exactly corrupt. It's just not complete. I think it's fine if the server sends a truncated array if the remainder is supposed to be NaN.